docs/multi-target-framework.md

```diff
@@ -0,0 +1,150 @@
+# Multi-targeting
+
+## Developer needs / scenarios
+
+* Compile against an earlier version of our bindings.
+    * Example: a customer could be producing a NuGet targeting .NET 7, and
+      want to support the initial release of .NET 7. Said customer must then
+      be able to build against the bindings we shipped at the time.
+    * Ref: https://github.com/dotnet/sdk/issues/30103
+* Consume preview packages for a preview version of Xcode.
+    * Ref: https://github.com/xamarin/xamarin-macios/issues/18343
+
+## Developer experience
+
+### TargetFramework=net7.0-ios16.4
+
+Builds with the bindings we released for iOS 16.4 (Xcode 14.3).
+
+### TargetFramework=net7.0-ios
+
+Builds with the default bindings. This can change in any release (this is
```

Comment:
This is a departure from the design for platforms in the TargetFramework.  Feel free to make this change if you think it's worth it, but here are some of the impacts:

- An update to the .NET SDK is more likely to break a project if it can change which version of the platform it's targeting.  We want to encourage people to use the latest SDK, so we try as best we can to avoid breaking changes.
- An update to the .NET SDK can mean that a package no longer supports being consumed by older SDKs.  This seems to run counter to the goal of fixing https://github.com/dotnet/sdk/issues/30103.  It would mean that package authors should probably always be explicit about the target platform version.

Reply:
> An update to the .NET SDK is more likely to break a project if it can change which version of the platform it's targeting. We want to encourage people to use the latest SDK, so we try as best we can to avoid breaking changes.

That's correct for probably every other platform but ours, because of how Xcode auto-updates: earlier workload versions may not work anymore with a new Xcode, so this can happen:

* Xcode auto-updates, and a developer's project doesn't build anymore. They try updating their workloads (still fails), so they try updating .NET (still same failure) - because we keep using the same old (broken) workload. They'll have to explicitly set the target platform version in the TargetFramework (and update it again the next time Xcode is auto-updated). Note that this will effect pretty much every project/developer, so after a while most, if not all, projects actively being developed will end up with an explicit target platform version.
* `dotnet new ...` would create a project that doesn't build by default. This can be mitigated if the templates put the OS version in the TargetFramework, but then you end up having the same issue for the next Xcode release: since the OS version is hardcoded, we keep trying to use the older workload, so it'll have to be continuously updated.

> An update to the .NET SDK can mean that a package no longer supports being consumed by older SDKs.

Not quite: package authors would have to be explicit about the target platform versions.

https://github.com/dotnet/sdk/issues/30103 is about making that possible (right now we're neither here nor there: we don't support building with older SDKs, neither explicitly nor implicitly).

Also note that package authors aren't often affected by the Xcode incompatibilities (they mostly show up when building apps, not packages).